MapboxNavigation/NavigationViewController.swift

```diff
@@ -696,6 +696,7 @@ extension NavigationViewController: NavigationServiceDelegate {
         if #available(iOS 14.0, *), accuracyAuthorization == .reducedAccuracy {
             let title = NSLocalizedString("ENABLE_PRECISE_LOCATION", bundle: .mapboxNavigation, value: "Enable precise location to navigate", comment: "Label indicating precise location is off and needs to be turned on to navigate")
             showStatus(title: title, spinner: false, duration: 20, animated: true, interactive: false)
+            mapView?.reducedAccuracyActivatedMode = true
         } else {
             //Fallback on earlier versions
```

Comment:
Currently `UserHaloCourseView` doesn't change its look when Precise Location if switched on. I think we should update it whenever `navigationServiceDidChangeAuthorization(_:didChangeAuthorizationFor:)` is called.

Reply:
you could add an observer to the notification that's posted whenever `navigationServiceDidChangeAuthorization(_:didChangeAuthorizationFor:)` is called:
https://github.com/mapbox/mapbox-navigation-ios/blob/e50fa9093da25b26030a2bdf10f29a6a98b7eea8/MapboxCoreNavigation/NavigationService.swift#L424-L427

Reply:
Thanks for the suggestion, I add the `mapView?.reducedAccuracyActivatedMode = false` when the Precise Location if switched on so right now it could switch between the two types of course view.